index.js

```diff
@@ -0,0 +1,143 @@
+        if (this.taskList[i].index === parentIndex + 1) {
+          if (this.taskList[i].completed === true) {
+            this.taskList[i].completed = false;
+            this.saveTasks();
+            return;
+          }
+          if (this.taskList[i].completed === false) {
+            this.taskList[i].completed = true;
+            this.saveTasks();
+          }
+        }
+      }
+    }
+
+    loadActivities = () => {
+      const activitiesListSection = document.getElementById('activities-list');
+      for (let i = 0; i < this.taskList.length; i += 1) {
+        const itemList = document.createElement('li');
+        itemList.id = i + 1;
+        itemList.innerHTML = this.displayWindow(this.taskList[i]);
+        activitiesListSection.appendChild(itemList);
+      }
+      this.loadCheckBoxes();
+    }
+
+    deleteTask = (taskId) => {
+      this.taskList.splice(taskId, 1);
+      for (let i = taskId; i < this.taskList.length; i += 1) {
+        this.taskList[i].index -= 1;
+      }
+      this.saveTasks();
+    }
+
+    upDateTask = (taskId, newDescription) => {
+      this.taskList[taskId].description = newDescription;
+      this.saveTasks();
+      window.location.reload();
+    }
+
+    saveTasks = () => {
+      localStorage.setItem(this.storageKey, JSON.stringify(this.taskList));
+    }
+}
+
+const taskManager = new TaskManager();
+taskManager.loadActivities();
+
+const activitiesInput = document.getElementById('activities-input');
+const addActivityBtn = document.getElementById('add-activity-btn');
+
+addActivityBtn.addEventListener('click', () => {
+  const activityValue = activitiesInput.value;
+  if (activityValue !== '') {
+    window.location.reload();
+    taskManager.createTask(activityValue);
+    activitiesInput.value = '';
+  }
+});
+
+activitiesInput.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    const activityValue = activitiesInput.value;
+    if (activityValue !== '') {
+      window.location.reload();
+      taskManager.createTask(activityValue);
+      activitiesInput.value = '';
+    }
+  }
+});
+
+const deleteButton = document.querySelectorAll('.trash');
+deleteButton.forEach((button) => {
+  button.addEventListener('click', (e) => {
+    const taskId = parseInt(e.target.parentElement.parentElement.id, 10);
+    taskManager.deleteTask(taskId);
+    window.location.reload();
+  });
+});
+
+const activityField = document.querySelectorAll('.li-activity');
+activityField.forEach((field, i) => {
+  field.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      const newDescription = e.target.value;
+      taskManager.upDateTask(i, newDescription);
+    }
+  });
+});
+
+const clearButton = document.getElementById('clear-button');
+clearButton.addEventListener('click', () => {
+  taskManager.taskList = taskManager.taskList.filter((task) => !task.completed);
+  taskManager.saveTasks();
+  window.location.reload();
+});
```

Comment:
- [ ] The code lacks comments that explain what each method or function does, which makes it difficult to understand for other developers.

- [ ] Variable names: Some variable and method names are not very descriptive and could be improved to make the code more readable.

Reply:
Thank you @federicaulzurrun, I appreciate your comments.